CommandLine/emake-tests/Parsing/parser-test.cpp

```diff
@@ -35,7 +35,244 @@ struct ParserTester {
 
 void assert_identifier_is(AST::Node *node, std::string_view name) {
   ASSERT_EQ(node->type, AST::NodeType::IDENTIFIER);
-  ASSERT_EQ(dynamic_cast<AST::IdentifierAccess *>(node)->name.content, name);
+  ASSERT_EQ(node->As<AST::IdentifierAccess>()->name.content, name);
+}
+
+std::string ExpectedMsg = "";
+
+MATCHER_P(IsDeclaration, decls, "") {
+  bool b1 = arg->type == AST::NodeType::DECLARATION;
+  if (!b1) {
+    ExpectedMsg = "From IsDeclaration Matcher: NodeType = DECLARATION\n";
+    *result_listener << "got NodeType = " << AST::NodeToString(arg->type) << "\n";
+    return 0;
+  }
+
+  auto *decl = arg->template As<AST::DeclarationStatement>();
```

Comment:
Now, there is no need to check whether this pointer is not equal to `nullptr`, as the previous condition is true.